### PR TITLE
feat(table): add toolbarActions to add actions to table toolbar or its overflow menu

### LINKED ISSUE #2951: [Table] toggle aggregation off-screen in RTL mode
### What package is this for?

- [X] React
- [ ] Angular

### Describe the bug

Screen Shot 2021-10-12 at 9 08 21 AM

### To Reproduce

1. https://next.carbon-addons-iot-react.com/?path=/story/1-watson-iot-table-table--basic-dumb-table
2. Turn on RTL
3. Open Overflow menu in Toolbar

### LINKED ISSUE #3063: [Table] Allow for more items to roll into table toolbar dropdown
As currently implemented we have the aggregations table toolbar action render in a dropdown. This renders only this item with no way to add to the dropdown. This should probably mirror how we do row actions where we provide a means to inform the table that the action is in a dropdown and we render it inside the dropdown that we manage. This would allow for custom actions that can live along side the aggregations. 

We need to support this in addition to the `view.toolbar.customToolbarContent` prop which is how we always advised for people to add items to the toolbar.

### PR DESCRIPTION
**Pull request by kevinsperrine**
Closes #3063 
Closes #2951 

**Summary**

You can add other actions to the toolbar or its overflow menu by using the `toolbarActions` prop on the view.toolbar. When clicking on an action it will trigger the `onApplyToolbarAction` callback passing the object used to create the action.
This object must contain an `id` and `labelText`, but it can also contain `hidden`, `disabled`, `isDelete`, `hasDivider`, or `renderIcon`, `isOverflow`, or `isActive` items.

**Change List (commits, features, bugs, etc)**

- add `toolbarActions` to `view.toolbar` table prop
- pass these actions to the TableToolbar using the same overflow menu used for aggregations (using `isOverflow`) or put them in the toolbar when not passing (`isOverflow`).
- if aggregations are enabled, any `toolbarActions` using `isOverflow` will be appended to the overflow menu
- if aggregations are not enabled, the `toolbarActions` using `isOverflow` will create the overflowMenu
- Any other `toolbarActions` not using `isOverflow` will be placed to the left of the overflow menu.

**Acceptance Test (how to verify the PR)**

- check the [basic dumb table story](https://deploy-preview-3076--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-table--basic-dumb-table) and turn on aggregations
- confirm the toolbar overflow menu contains other items.
- Edit should be disabled
- Delete should be red when hovering
- there should not be one named `Hidden` even though it's in the actions.
- when clicking on `Toggle something`
- confirm the `onApplyToolbarAction` action is triggered in storybook
- https://deploy-preview-3076--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-table--with-filters
- should show the "eye-off" icon in the toolbar and trigger the `onApplyToolbarAction` action in storybook when clicked
**Regression Test (how to make sure this PR doesn't break old functionality)**

- confirm this doesn't adversely affect the aggregations in any way.

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

**Review by stuckless — APPROVED** [review]

File: packages/react/src/components/Table/TableToolbar/TableToolbar.jsx [hunk 1]
```diff
@@ -227,13 +233,49 @@ const TableToolbar = ({
...
+        <OverflowMenuItem
+          data-testid={`${testID || testId}-toolbar-overflow-menu-item-${action.id}`}
+          itemText={action.itemText}
+          key={`table-aggregations-overflow-item-${action.id}`}
+          onClick={() => {
+            trackOpenState();
+            onApplyExtraAction(action);
+          }}
+          disabled={isDisabled || action.disabled}
+          isDivider={action.isDivider}
```
[the remark this answers is not in the document]
Comment (kevinsperrine) [reply packages/react/src/components/Table/TableToolbar/TableToolbar.jsx:273]: Thanks, @stuckless. I realized that this morning. 1) The `rowActions` use different prop names (and support more features), so I'm going to rework this to align those the two better. 2) I'm second guessing the naming of `extraActions`. Initially, I based it on the `extraActions` added to the cards, but I'm wondering if `toolbarActions` fits better for the table (and aligns better when compared to `rowActions`. Do you have any thoughts?

[the remark this answers is not in the document]
Comment (bjornalm) [reply packages/react/src/components/Table/TableToolbar/TableToolbar.jsx]: `toolbarActions` is better. The table row actions don't have to be in an overflow, but these ones must, so perhaps we should call these `toolbarOverflowActions` or something?

[the remark this answers is not in the document]
Comment (stuckless) [reply packages/react/src/components/Table/TableToolbar/TableToolbar.jsx]: My personal thought is that it seems silly to have actions physically separated into 2 separate configurations.  I'd rather see just some "toolbarActions" with a prop that identifies if the action is overflow or not.  My reasoning for this is that typically when you are defining actions, you might not always know where it should rendered, and then later someone says, hey just move it into to overflow.  This becomes trivial with just changing a prop vs now needing to find/configure a new structure with new callbacks, etc, just to visually render it somewhere else.

I agree that the toolbarActions (or toolbarOverflowActions if we must) better describes what these are.

[the remark this answers is not in the document]
Comment (kevinsperrine) [reply packages/react/src/components/Table/TableToolbar/TableToolbar.jsx]: Thanks, @bjornalm and @stuckless. I've changed the name to `toolbarActions` (the callback to `onApplyToolbarAction` and added the `isOverflow` prop to put items into the overflow menu, and in the toolbar if not.

**Review by davidicus — APPROVED** [review]

**Pull request merged**